### PR TITLE
Add an audit log

### PR DESCRIPTION
**Pull request by dfellis**
- Add the new 'iasql_audit_log' table to the platform and fix the module migration generation to require zero config changes to run
- Write the audit trigger function and attach/detach it on install/uninstall
- Syntax fixes for the trigger function

### CODE REVIEW

**Comment by dfellis** [comment]
Screenshot from 2022-05-12 18-03-16

Looks like this. Tracks all DB changes, when they happened, and the user that did it (the engine's user would indicate `sync` and `install` changes, while the "normal" user would be changes to be pushed at the cloud.

File: src/modules/0.0.4/iasql_platform/entity/index.ts [hunk 1]
```diff
@@ -45,4 +46,42 @@ export class IasqlTables {
...
+
+@Entity()
+export class IasqlAuditLog {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({
+    type: 'timestamp without time zone',
+  })
+  ts: Date;
```
Comment (aguillenv) [inline src/modules/0.0.4/iasql_platform/entity/index.ts:65]: should this be with timezone to save it in UTC?

File: src/services/iasql.ts [hunk 2]
```diff
@@ -870,6 +870,16 @@ ${Object.keys(tableCollisions)
         return mt;
       }) ?? [];
       await orm.save(Modules.IasqlPlatform.utils.IasqlTables, modTables);
+      // For each table, we need to attach the audit log trigger, if the platform is >=0.0.4
+      if (!['v0_0_1', 'v0_0_2', 'v0_0_3'].includes(versionString)) {
+        for (const table of md.provides.tables) {
+          await queryRunner.query(`
+            CREATE TRIGGER ${table}_audit
+            AFTER INSERT OR UPDATE OR DELETE ON ${table}
+            FOR EACH ROW EXECUTE FUNCTION iasql_audit();
```
Comment (aguillenv) [inline src/services/iasql.ts:879]: should this be the deferred one?

File: src/modules/0.0.4/iasql_functions/sql/create_fns.sql [hunk 3]
```diff
@@ -1,3 +1,22 @@
+-- TODO: Does this belong here or in a similar file in the iasql_platform module?
+create or replace function iasql_audit() returns trigger
+language plpgsql security definer
+as $$
+begin
+  if (TG_OP = 'INSERT') then
```
Comment (aguillenv) [inline src/modules/0.0.4/iasql_functions/sql/create_fns.sql:6]: does the `TG_OP` is a Postgres standard to define the trigger operation or where does it come from?

File: src/services/iasql.ts (hunk 2 above)
Comment (aguillenv) [reply src/services/iasql.ts:879]: Oh that quote answer my concern. Perfect!

**Review by depombo — APPROVED** [review]

**Pull request merged**